runtime/parachains/src/configuration.rs

```diff
@@ -284,6 +284,18 @@ pub mod pallet {
 	pub trait Config: frame_system::Config + shared::Config {
 		/// Weight information for extrinsics in this pallet.
 		type WeightInfo: WeightInfo;
+
+		/// Maximum value that `hrmp_max_parachain_outbound_channels` or
+		/// `hrmp_max_parathread_outbound_channels` can be set to.
+		///
+		/// This is used for benchmarking sanely bounding relevant storage items.
+		type HrmpMaxOutboundChannelsBound: Get<u32>;
```

Comment:
Long term, I don't think this patter is good. We probably will need to add a lot of bound types here, one for each field of `HostConfigurations`. I don't see it within the scope of this PR, but if we go down that road, it is probably worthwhile to abstract all of these types into one new type, maybe called `struct HostConfigurationBounds` and only use that to minimize the footprint in the interface.

Reply:
Also, note that for simplicity, for now I have combined the parachain and parathread variant. This might help us use `BoundedVec` in some places as well, but that didn't work out.